app/javascript/gamesFactory.js

```diff
@@ -0,0 +1,226 @@
+/* global angular, inArray, escape */
+import {web3, Chess} from '../../contract/Chess.sol';
+angular.module('dappChess').factory('games', function (navigation, $rootScope, $route) {
+  const games = {
+    list: [],
+    openGames: []
+  };
+  /*
+   * Structure of games list:
+   * [
+   *  gameId: <string>,
+   *  {
+   *    self: {
+   *      username: <string>,
+   *      accountId: <string>,
+   *      color: <string>
+   *    },
+   *    opponent: {
+   *      username: <string>,
+   *      accountId: <string>,
+   *      color: <string>
+   *    }
+   *  }
+   * ]
+   */
+
+  games.getGame = function (id) {
+    return games.list.find(function (game) {
+      return game.gameId === id;
+    });
+  };
+
+  /**
+   * Convert an array to a game object as seen in the contract with the given gameId.
+   * @param gameId of the game
+   * @param array containing the data
+   * @returns object as seen in the contract
+   */
+  games.convertArrayToGame = function (gameId, array) {
+    return {
+      gameId: gameId,
+      player1: array[0],
+      player2: array[1],
+      player1Alias: array[2],
+      player2Alias: array[3],
+      nextPlayer: array[4],
+      playerWhite: array[5],
+      winner: array[6],
+      state: array[7]
+    };
+  };
+
+  /**
+   * Add a game to the list, if there is no game with the same id.
+   * @param contractGameObject An object with the structure of a Game in the contract<br>
+   *     { player1: "<address>", player2: "<address>", ... }
+   * @returns the new created game or undefined, if the gameId is already in the list
+   */
+  games.add = function (contractGameObject) {
+    if (typeof games.getGame(contractGameObject.gameId) !== 'undefined') {
+      return;
+    }
+    let game = { gameId: contractGameObject.gameId };
+
+    if (inArray(contractGameObject.p2accountId, web3.eth.accounts)) {
```

Comment:
As for the tests, I think we din't really decide on frontend tests. We could add protractor end-to-end tests, but I'd first want to figure out how usable they are and how much added work that is.